src/solvers/smt2/smt2_parser.h

```diff
@@ -46,6 +46,11 @@ class smt2_parsert
 
   struct named_termt
   {
+    named_termt(const exprt &_term, const symbol_exprt &_name)
+      : term(_term), name(_name)
+    {
+    }
+
```

Comment:
should have been in a separate commit to explain why this change is needed

Reply:
+1 this is not clear.

Reply:
I have added a comment (default-constructing a `symbol_exprt` is deprecated, thus the `name` member needs to be initialised upon construction). Goes together with the use of `emplace` in smt2_parser.cpp.